src/csharp/Grpc.Tools.Tests/ProtoCompileCommandLineGeneratorTest.cs

```diff
@@ -175,5 +175,86 @@ public void DirectorySlashTrimmingCases(string given, string expect)
             Assert.That(_task.LastResponseFile,
                         Does.Contain("--csharp_out=" + expect));
         }
+
+        [TestCase(
+            "../Protos/greet.proto(19) : warning in column=5 : warning : When enum name is stripped and label is PascalCased (Zero) this value label conflicts with Zero.",
+            "../Protos/greet.proto",
+            19,
+            5,
+            "warning : When enum name is stripped and label is PascalCased (Zero) this value label conflicts with Zero.")]
```

Comment:
nit: Is it possible to trim out the "warning : "? The warning icon already conveys that info.

Could either by done via regex, or some post regex text processing:

```cs
string text = match.Groups["TEXT"].Value;
if (text.StartsWith("warning : "))
{
     text = text.Substring(10);
}
```

Are there any errors prefixed with "error : "?

Reply:
I found it not very standardized. The relevant logic is here:
https://github.com/protocolbuffers/protobuf/blob/master/src/google/protobuf/compiler/command_line_interface.cc#L353-L357

However, it's not entirely consistent what happens since the message might be prefixed with `warning : `. I was hoping to keep the logic as simple as possible and didn't want to modify the message in any way. I suppose removing that specific prefix might be safe though.

Reply:
thoughts @jtattermusch ?

Reply:
I have the changes at https://github.com/JunTaoLuo/grpc/commit/4f8e22f9445631fe878380a5c13a30934a2340a8. But to get that merged to this PR requires another run of tests. @jtattermusch to decide whether we should merge now as is or incorporate the changes and wait for tests to run.

Reply:
I think it makes sense to merge as is and keep improving iteratively?  Warning messages starting with "warning :" prefix doesn't seem as a big problem to me. Being able to jump to the right location is much more important.

Reply:
Ok let's merge as is. Will file issue to follow up.

Reply:
Filed https://github.com/grpc/grpc/issues/18518 to track.